src/base/bittorrent/session.cpp

```diff
@@ -1931,11 +1948,20 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
     if (!fromMagnetUri) {
         if (params.restored) {
             // Set torrent fast resume data
-            p.resume_data = {fastresumeData.constData(), fastresumeData.constData() + fastresumeData.size()};
+            p.resume_data = std::vector<char> {fastresumeData.constData(), fastresumeData.constData() + fastresumeData.size()};
             p.flags |= libt::add_torrent_params::flag_use_resume_save_path;
         }
         else {
-            p.file_priorities = {params.filePriorities.begin(), params.filePriorities.end()};
+            std::transform(params.filePriorities.cbegin(), params.filePriorities.cend()
+                           , std::back_inserter(p.file_priorities), [](DownloadPriority priority)
+            {
+#if (LIBTORRENT_VERSION_NUM < 10200)
+                return static_cast<boost::uint8_t>(priority);
+#else
+                return static_cast<lt::download_priority_t>(
+                            static_cast<lt::download_priority_t::underlying_type>(priority));
```

Comment:
seems weird that casting to `lt::download_priority_t::underlying_type` is not enough and it needs a second cast, is it a good idea to make a suggestion to libt author?

Reply:
>is it a good idea to make a suggestion to libt author?

I don't think so. I've discussed this with him before. "Using strong types for everything" is presented as a feature. But I don't think it's a good idea in General, so I preferred to narrow it down and keep use simple integers for indices, flags etc. as it was before.
The only exception is DownloadPriority I use `enum class` for. So it requires two-step cast (`enum class` -> `simple integer type` -> `libtorrent strong type`). But we can use simple integer type instead of `enum class` for DownloadPriority if you want (enum is used currently to declare some predefined priority constants).

Reply:
OK, please don't bother, just keep it as is.